modules/local/misc.nf

```diff
@@ -0,0 +1,117 @@
+include { saveFiles; getSoftwareName } from './functions'
+
+process CAT_FASTQ {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "conda-forge::pigz=2.6" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/mulled-v2-2b04072095278721dc9a5772e61e406f399b6030:7c7abf911e92d7fb831611ffb965f3cf7fe2c01d-0"
+    } else {
+        container "quay.io/biocontainers/mulled-v2-2b04072095278721dc9a5772e61e406f399b6030:7c7abf911e92d7fb831611ffb965f3cf7fe2c01d-0"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*.fastq.gz"), emit: reads
+    path "versions.yml" , emit: versions
+
+    script:
+    """
+    fastqFile=\$(ls $reads | head -1)
+    if grep -q "fastq.gz" <<< "\$fastqFile"; then
```

Comment:
With `head -1`, you could end up checking a random non-fastq file you don't want to check. This also assumes that `reads` is a single directory rather than a list of directories or files or both.

File extensions could sometimes be `.fq.gz` for compressed FASTQ or `.fq` for uncompressed. So the grep regex command should be `grep -qE "\.(fastq|fq)\.gz$"`

You could use the Linux `file` command to check the file type as well:
```bash
$ file -L --mime-type some-file
some-file: application/gzip
```

Reply:
Updated ad we discuss via email